src/main/scala/com/cognite/spark/datasource/CdpRelation.scala

```diff
@@ -18,6 +18,14 @@ object Setter {
       case _ => Some(new Setter(set.get, false))
     }
 }
+case class NonNullableSetter[A](set: A)
+object NonNullableSetter {
+  def apply[A](set: Option[A]): Option[NonNullableSetter[A]] =
+    set match {
```

Comment:
I think this is `set.map(NonNullableSetter(_))`? In which case we may not need the object and the apply method, seems like a small benefit compared to just doing `set.map(NonNullableSetter(_))` instead of `NonNullableSetter(map)`.

Reply:
Good catch, was caught up in the thinking from Setter here.

Reply:
This is still relevant it seems. :)

Reply:
Good catch again!